src/memray/_memray/source.cpp

```diff
@@ -25,22 +25,32 @@ namespace memray::io {
 FileSource::FileSource(const std::string& file_name)
 : d_file_name(file_name)
 {
-    d_stream.open(d_file_name, std::ios::binary | std::ios::in);
-    if (!d_stream) {
+    d_raw_stream.open(d_file_name, std::ios::binary | std::ios::in);
+    if (!d_raw_stream) {
         throw IoError{"Could not open file " + file_name + ": " + std::string(strerror(errno))};
     }
+    uint32_t header;
+    d_raw_stream.read(reinterpret_cast<char*>(&header), 4);
+    d_raw_stream.seekg(0, std::ios::beg);
+
+    if (header == 0x184D2204) {
```

Comment:
Instead of reading into a `uint32_t`, it would be better to read into a `char[4]` so that we're endianness agnostic. Or alternatively, check for `htonl(header) == 0x04224D18` I suppose.